pkg/placement/placement.go

```diff
@@ -266,12 +268,12 @@ func (p *Service) ReportDaprStatus(stream placementv1pb.Placement_ReportDaprStat
 
 	clientID, err := p.validateClient(stream)
 	if err != nil {
-		return err
+		return apiErrors.PlacementServiceInternalError(fmt.Sprintf("internal or unauthenticated error: %s", err))
 	}
 
 	firstMessage, err := p.receiveAndValidateFirstMessage(stream, clientID)
 	if err != nil {
-		return err
+		return apiErrors.PlacementServicePermissionDenied(fmt.Sprintf("internal of permission denied error: %s", err))
```

Comment:
Same here. We should respect the error `receiveAndValidateFirstMessage` returns.